### PR TITLE
update reshape spec

### PR DESCRIPTION
**Pull request by pelszkow**
### Details:
 - *update reshape spec*

### Tickets:
 - *45954*

### CODE REVIEW

File: docs/ops/shape/Reshape_1.md [hunk 1]
```diff
@@ -8,40 +8,41 @@
...
+If `special_zero` is set to `true` index of `0` cannot be larger than the rank of the input tensor.
 
 **Attributes**:
 
 * *special_zero*
 
-  * **Description**: *special_zero* controls how zero values in `shape` are interpreted. If *special_zero* is `false`, then 0 is interpreted as-is which means that output shape will contain a zero dimension at the specified location. Input and output tensors are empty in this case. If *special_zero* is `true`, then all zeros in `shape` implies the copying of corresponding dimensions from `data.shape` into the output shape *(left aligned)*.
+  * **Description**: *special_zero* controls how zero values in `shape` are interpreted. If *special_zero* is `false`, then `0` is interpreted as-is which means that output shape will contain a zero dimension at the specified location. Input and output tensors are empty in this case. If *special_zero* is `true`, then all zeros in `shape` implies the copying of corresponding dimensions from `data.shape` into the output shape *(left aligned)*.
   * **Range of values**: `false` or `true`
   * **Type**: boolean
```
Comment (jdanieck) [inline docs/ops/shape/Reshape_1.md:23]: Recently we tend to enclose type in backticks:  `boolean`

File: docs/ops/shape/Reshape_1.md [hunk 2]
```diff
@@ -8,40 +8,41 @@
...
   * **Default value**: None
   * **Required**: *yes*
 
 **Inputs**:
 
-*   **1**: `data` -- multidimensional input tensor of type *T*. *Required*.
+*   **1**: `data` - multidimensional input tensor of type *T*. **Required**.
```
Comment (jdanieck) [inline docs/ops/shape/Reshape_1.md:29]: Recently instead of "_multidimensional input tensor of type *T*_", we write "_A tensor of type T and arbitrary shape_" to be explicit about the shape

**Review by jdanieck — APPROVED** [review]

File: docs/ops/shape/Reshape_1.md (hunk 1 above)
Comment (pelszkow) [reply docs/ops/shape/Reshape_1.md]: updated.

**Comment by pelszkow** [comment]
@lazarevevgeny, will you find time to review?  Thanks.

**Review by ggalieroc — APPROVED** [review]

File: docs/ops/shape/Reshape_1.md [hunk 3]
```diff
@@ -8,40 +8,41 @@
...
   * **Default value**: None
   * **Required**: *yes*
 
 **Inputs**:
 
-*   **1**: `data` -- multidimensional input tensor of type *T*. *Required*.
+*   **1**: `data` a tensor of type T and arbitrary shape. **Required**.
 
-*   **2**: `shape` -- 1D tensor of type *T_SHAPE* describing output shape. *Required*.
+*   **2**: `shape` 1D tensor of type *T_INT* describing output shape. **Required**.
```
Comment (lazarevevgeny) [inline docs/ops/shape/Reshape_1.md:31]: May be it is my preference but I think that the name `T_SHAPE` has more meaning than `T_INT`. So it is better to revert this change.

Comment (lazarevevgeny) [reply docs/ops/shape/Reshape_1.md:31]: @pelszkow, do you plan to resolve this comment?

Comment (pelszkow) [reply docs/ops/shape/Reshape_1.md]: fix

**Review by ggalieroc — APPROVED** [review]

**Review by jdanieck — APPROVED** [review]
